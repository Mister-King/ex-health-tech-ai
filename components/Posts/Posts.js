/**
 * Posts component
 */

import React from 'react';
import { View, FlatList } from 'react-native';
import PostItem from './PostItem/PostItem';

import styles from './_posts.scss';
import ListDivider from '../ListDivider/ListDivider';

const Posts = props => {
    const {
        posts,
        updateActivePost,
        navigation,
    } = props;

    // Internal Functions
    const renderItem = ({ item }) => (
        <PostItem
            title={item.title}
            id={item.id}
            updateActivePost={updateActivePost}
            navigation={navigation}
        />
    );

    // Render
    return (
        <View>
            <FlatList
                data={posts}
                renderItem={renderItem}
                keyExtractor={item => item.id.toString()}
                className={styles.postlist}
                ItemSeparatorComponent={() => <ListDivider />}
            />
        </View>
    );
};

export default Posts;
