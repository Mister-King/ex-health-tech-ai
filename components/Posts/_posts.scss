/**
 * Posts
 */

.postlist {
	text-align: left;
	padding: 0 20px;
}
