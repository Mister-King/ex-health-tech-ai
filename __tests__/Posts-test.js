/* eslint no-undef: 0 */
/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Posts from '../components/Posts/Posts';
import { PostsData } from '../jest/jest.data';

test('renders list of post titles', () => {
    const tree = renderer.create(
        <Posts posts={PostsData} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
