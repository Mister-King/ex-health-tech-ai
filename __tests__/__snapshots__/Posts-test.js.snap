// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of post titles 1`] = `
<View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
          "id": 1,
          "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
          "userId": 1,
        },
        Object {
          "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
          "id": 2,
          "title": "qui est esse",
          "userId": 1,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style="postlist"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style="postitem"
        >
          <Text
            style="postitem__text"
          >
            sunt aut facere repellat provident occaecati excepturi optio reprehenderit
          </Text>
        </View>
        <View
          style="divider"
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style="postitem"
        >
          <Text
            style="postitem__text"
          >
            qui est esse
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
